### PR TITLE
New rule to remove unnecessary separators and dots

### LINKED ISSUE #12166: [Feature Request] New rule to remove unnecessary separators and dots
Many methods contain tabs or spaces at the end of each code line. Also some final dot that is not necessary.

We should have a lint rule detecting such methods together with an autofix option

### PR DESCRIPTION
**Pull request by astares**
Fix #12166

### CODE REVIEW

**Comment by astares** [comment]
image

**Comment by astares** [comment]
CI issue on Windows is unrelated

File: src/Renraku/ReCompactSourceCodeCritic.class.st [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"
+A code critique to take care of unnecessary seperators at end of source code lines and unnecessary dot at end of method
```
Comment (vonbecmann) [inline src/Renraku/ReCompactSourceCodeCritic.class.st]: separators is misspelled

Comment (astares) [reply src/Renraku/ReCompactSourceCodeCritic.class.st]: fixed

**Review by vonbecmann — APPROVED** [review]

**Review by vonbecmann — APPROVED** [review]

**Comment by Ducasse** [comment]
I like that it fixes the problem and do it without reformatting the code :) super cool.

**Comment by demarey** [comment]
remove unnecessary separators is really nice!
On the other hand, I do not like removing trailing dot. All expressions can end with a dot. Removing the last dot in a method forces the writer to remember to add a missing dot before adding new expressions to a method. It is easier to always put the dot.

**Comment by astares** [comment]
@demarey 

1. If you do not like it, then ban the rule for your package. Simple.

2. But finishing with a dot every method just for the case later there might eventually (who really knows?) be another expression I would call "premature dotting". 

See a detailed discussion on this in the thread:   https://twitter.com/TorstenAstares/status/1609666189382254593

**Pull request merged**